src/sentry/api/endpoints/project_rules.py

```diff
@@ -825,11 +825,12 @@ def post(self, request: Request, project) -> Response:
         owner = data.get("owner")
         if owner:
             try:
-                # TODO(mark) Use owner.resolve() intead when owner is removed.
-                actor = owner.resolve_to_actor()
-                kwargs["owner"] = actor.id
-                kwargs["owner_user_id"] = actor.user_id
-                kwargs["owner_team_id"] = actor.team_id
+                kwargs["owner_user_id"] = None
+                kwargs["owner_team_id"] = None
+                if owner.type == User:
+                    kwargs["owner_user_id"] = owner.id
+                if owner.type == Team:
+                    kwargs["owner_team_id"] = owner.id
```

Comment:
I would like to revisit `ActorTuple` in the future. Having models as properties feels like an awkward solution. I think we might be better off with an enum.

Reply:
Completely agree